cpp/src/arrow/pretty_print.cc

```diff
@@ -145,7 +145,8 @@ class ArrayPrinter : public PrettyPrinter {
     int window = is_container ? options_.container_window : options_.window;
     for (int64_t i = 0; i < array.length(); ++i) {
       const bool is_last = (i == array.length() - 1);
-      if ((i >= window) && (i < (array.length() - window))) {
```

Comment:
Wouldn't the same effect be achieved by just replacing `i >= window` with `i > window`?

Reply:
I don't think so, because if you have (say) 6 elements with a window of 2, then you'd render 5 of the elements when you only mean to render 4, right?

Reply:
That said, we should comment this check to be clear about what it's achieving

Reply:
> I don't think so, because if you have (say) 6 elements with a window of 2, then you'd render 5 of the elements when you only mean to render 4, right?

You're right :)